firmware/library/L1_Drivers/dac.hpp

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <cstdint>
+#include <cstdio>
```

Comment:
It doesn't look like you are using cstdio. Therefore, you should remove this include.